tests/publisher_spec.rb

```diff
@@ -0,0 +1,37 @@
+require 'rspec'
+require 'simplecov'
+SimpleCov.start
+require_relative '../lib/publisher'
+require_relative '../lib/subscriber'
+
+
+RSpec.describe Subscriber do
+  context '#valid_queue_name?' do
+    let (:valid_queue_name) { "test_123" }
+    
+    it "should validate the url's per aws sqs format" do
+      expect(Publisher.valid_queue_name?(valid_queue_name)).to be_truthy
```

Comment:
Wasn't there a validation required for the length of the queue name?
Make the changes and test a valid and invalid queue name.

Reply:
Length of the queue already part of the regex. {3,80}. So that has been taken care of.

Reply:
The test for validation of the queue_name is pending.

Reply:
@glen its there in publisher_spec.rb:L9.. The title of the test is wrong. Correcting it now.

Reply:
Meant that you also need to have a `be_falsey` test. So test with an invalid queue_name

Reply:
in case of invalid queue names, we raise error. Have added checks for those.